tests/15_Versioning/VersionManagerTest.php

```diff
@@ -275,6 +283,39 @@ public function testRestorePendingChanges()
         $this->vm->restore(true, $version);
     }
 
+    public function testRestoreIsCheckedIn()
+    {
+        $nodePath = '/tests_version_base/versioned';
+        $version = $this->vm->checkpoint($nodePath);
+        $this->vm->checkout($nodePath);
+        $this->assertTrue($this->vm->isCheckedOut($nodePath));
+
+        $this->vm->restore(true, $version);
+        $this->assertFalse($this->vm->isCheckedOut($nodePath));
+    }
+
+    public function testRestoreBaseProperties()
+    {
+        // TODO also check for primary node type and uuid once it can be changed
```

Comment:
i would propose we assume the uuid immutable and you can add a check
that throws an exception when the frozen uuid is not the one of the
current node, saying something unexpected happened...

for mixins, they certainly can change.

about the primary type, this seems to be another jackrabbit restriction
that is actually a mistake https://github.com/jackalope/jackalope/issues/247

but in my opinion we can just leave a TODO here referencing that issue,
and a comment in that issue that it would also have to be fixed in
version restore if this is ever changed. as its now, its not possible to
change the primary type so no need (yet) to restore it.